### PR TITLE
Document TypeScript compiler options

### PR DESCRIPTION
**Pull request by Soremwar**
First PR 😃 

This PR adds information on how to use `tsconfig.json` inside Deno projects, the default values set for this settings and a list of the options currently supported by Deno.

**Note**: This list doesn't showcase `"lib"` options as configurable due to issues with this configuration being reported on #4234.

### CODE REVIEW

File: std/manual.md [hunk 1]
```diff
@@ -1210,6 +1210,61 @@ console.log(result["/foo.ts"].map);
 We would expect the `enum` would be rewritten to an IIFE which constructs the
 enumerable, and the map to be defined.
 
+## TypeScript Compiler Options
+
+In Deno ecosystem, all strict flags are enabled in order to comply with TypeScript
```
Comment (kitsonk) [inline std/manual.md]: I think you will need to format this `./tools/format.py`

File: std/manual.md [hunk 2]
```diff
@@ -1210,6 +1210,61 @@ console.log(result["/foo.ts"].map);
...
+program execution.
+
+You do need to explicitly tell Deno where to look for this configuration, in order to do so
+you can use the `-c` argument when executing your application.
+
+```bash
+deno -c tsconfig.json mod.ts
+```
+
+Currently allowed settings, as well as their default values in Deno go as follows:
```
Comment (kitsonk) [inline std/manual.md]: Might be good to link to https://www.typescriptlang.org/docs/handbook/compiler-options.html somewhere here.

File: std/manual.md [hunk 3]
```diff
@@ -1210,6 +1210,61 @@ console.log(result["/foo.ts"].map);
...
+
+```json
+{
+   "compilerOptions": {
+      "allowJs": false,
+      "allowUmdGlobalAccess": false,
+      "allowUnreachableCode": false,
+      "allowUnusedLabels": false,
+      "alwaysStrict": true,
+      "assumeChangesOnlyAffectDirectDependencies": false,
```
Comment (kitsonk) [inline std/manual.md]: Hmmm... there are some new compiler options here that I need to have a look at, like this one.  I am not sure we have the right settings, and they should be hard-coded and not left up to manipulation.

File: std/manual.md (hunk 1 above)
Comment (Soremwar) [reply std/manual.md]: I thought that it would not affect Markdown, but CI already failed because of this😓

File: std/manual.md (hunk 3 above)
Comment (Soremwar) [reply std/manual.md]: Yep. In our Gitter discussion earlier you mentioned the non-negotiable options Deno ignores to avoid breaking things.

So I removed those options and compared with Typescript docs to check the default values of the remaining ones.

In order to check the possible impact of the new flags, we could set them to the alternative of the default and run the tests(chances are its way more complex than this and we end up adding test cases for these edge cases)

Comment (kitsonk) [reply std/manual.md]: I think having them in here is great now, because if we "hard code" them in the future, we can reflect the changes here as well.  Thanks for the effort.

File: std/manual.md (hunk 2 above)
Comment (Soremwar) [reply std/manual.md]: Solved

Comment (Soremwar) [reply std/manual.md]: Added

File: std/manual.md (hunk 1 above)
Comment (Soremwar) [reply std/manual.md]: Formatted

File: std/manual.md (hunk 3 above)
Comment (Soremwar) [reply std/manual.md]: All ok @kitsonk! Added "**lib**" as an options since #4234 was resolved.

**Comment by David-Else** [comment]
I see these were added to `Currently allowed settings` as default values in this commit:
```
   "strict": true,
    "strictBindCallApply": true,
    "strictFunctionTypes": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "alwaysStrict": true
```
...but if `strict` is set to true, then the other options are turned on **automatically**.

You can validate it in the official v2 docs which are not out yet:

https://www.typescriptlang.org/v2/tsconfig#strict

If deno actually turns them all to `true` when `strict` is `true`, then it is redundant and confusing. The new docs are not really clear about the way `strict` works which could cause problems. I am pretty sure it is best practice to turn `strict` on and then turn any off you don't want on an individual basis.

**Comment by Soremwar** [comment]
> If deno actually turns them all to `true` when `strict` is `true`, then it is redundant and confusing. The new docs in this commit are not really clear about the way `strict` works which could cause problems. I am pretty sure it is best practice to turn `strict` on and then turn any rules off you don't want on an individual basis.

You are right, they are not enabled by Deno, but by turning on `strict` on the Typescript options used internally on Deno.

However I needed to list all possible options you could use on Deno, so they had to be in this document, even if `strict` implicitly turns them on.

PD: If you turn them off in your personal configuration, they will be excepted from the `strict` setting.
